src/view/dataElements/xdmObject.jsx

```diff
@@ -79,45 +113,102 @@ const XdmObject = ({ initInfo, formikProps, schema }) => {
 XdmObject.propTypes = {
   initInfo: PropTypes.object.isRequired,
   formikProps: PropTypes.object.isRequired,
-  schema: PropTypes.object
+  schemasMeta: PropTypes.array,
+  selectedSchemaMeta: PropTypes.object,
+  setSelectedSchemaMeta: PropTypes.func
 };
 
 const XdmExtensionView = () => {
-  const [schema, setSchema] = useState();
+  const [schemasMeta, setSchemasMeta] = useState();
+  const [selectedSchemaMeta, setSelectedSchemaMeta] = useState();
 
   return (
     <ExtensionView
       getInitialValues={({ initInfo }) => {
-        return fetchSchema({
-          edgeConfigId: initInfo.extensionSettings.instances[0].edgeConfigId,
+        // TODO: If we've already loaded the schemas, then don't load the
+        // schemas again. Just call getInitialValues with the newly
+        // selected schema and return the result.
+        return fetchSchemaList({
           orgId: initInfo.company.orgId,
           imsAccess: initInfo.tokens.imsAccess
-        }).then(fetchedSchema => {
-          setSchema(fetchedSchema);
+        })
+          .then(_schemasMeta => {
+            setSchemasMeta(_schemasMeta);
 
-          if (!fetchedSchema) {
-            return {};
-          }
+            if (!_schemasMeta.length) {
+              return {};
+            }
 
-          const initialValues = getInitialValues({
-            settings: initInfo.settings,
-            schema: fetchedSchema
-          });
+            // TODO: don't initialize form state if the schema doesn't match
+            let existingSchema;
```

Comment:
Maybe rename to `existingSchemaMeta` or, more explicitly, `schemaMetaMatchingPreviousSave`. The main point is if it's a schema meta instead of a full schema, then it's probably important to keep the "meta" part in the name.